src/lib/Libcmds/pbs_json.c

```diff
@@ -151,6 +168,11 @@ strdup_escape(JsonEscapeType esc_type, const char *str)
 				break;
 			case '"':
 				if (esc_type == JSON_ESCAPE) {
+					bufstr = str;
+					bufstr--;
```

Comment:
Is there any risk of `bufstr` running out of boundary?

Reply:
@minghui-liu : I'm checking for that condition immediately after this, before looking at content of bufstr:
```
173               if ((bufstr >= orig_str) && (*bufstr != '\\')) {
174		          buf[i++] = '\\';
175		     }
```